pkg/arvo/app/landscape.hoon

```diff
@@ -41,68 +46,155 @@
   ^-  (quip card _this)
   |^
   ?+  mark  (on-poke:def mark vase)
+      %landscape-action
+    (landscape-action !<(action vase))
+  ::
       %handle-http-request
     =+  !<([id=@ta req=inbound-request:eyre] vase)
     :_  this
-    %+  give-simple-payload:app    id
-    %+  require-authorization:app  req
-    handle-http-request
+    %+  give-simple-payload:app  id
+    (handle-http-request req)
   ==
   ::
+  ++  landscape-action
+    |=  act=action
+    ^-  (quip card _this)
+    ?-  -.act
+        %serve-dir
+      :-  ~
+      this(serving (~(put by serving) url-base.act clay-base.act))
+        %unserve-dir
+      :-  ~
+      this(serving (~(del by serving) url-base.act))
+    ==
+  ::
   ++  handle-http-request
     |=  =inbound-request:eyre
     ^-  simple-payload:http
-    ?.  =(src.bowl our.bowl)  [[403 ~] ~]
-    =*  req  request.inbound-request
+    =*  req       request.inbound-request
+    =*  headers   header-list.req
     =/  req-line  (parse-request-line url.req)
-    ?+  method.req  not-found:gen
-        %'GET'  (handle-get header-list.req req-line)
-    ==
-  ::
-  ++  handle-get
-    |=  [headers=header-list:http req-line=request-line]
-    ^-  simple-payload:http
-    ?~  ext.req-line
-      $(req-line [[[~ %html] (snoc site.req-line 'index')] args.req-line])
-    ?~  site.req-line  not-found:gen
+    ?.  =(method.req %'GET')
+      not-found:gen
+    =?  req-line  ?=(~ ext.req-line)
+      [[[~ %html] (snoc site.req-line 'index')] args.req-line]
+    ?>  ?=(^ ext.req-line)
+    ?~  site.req-line
+      not-found:gen
     =*  url-prefix  url-prefix.configuration
     ?.  ?|  ?&(?=(^ url-prefix) =((need url-prefix) i.site.req-line))
             =(url-prefix ~)
         ==
       not-found:gen
-    =/  file=(unit octs)
-      ?+  site.req-line  ~
-          [%'' %'index' ~]  (get-file-at /app/landscape [t.site u.ext]:req-line)
-          [%'~link' *]      (get-file-at /app/landscape [t.site u.ext]:req-line)
-          [%'~chat' *]      (get-file-at /app/landscape [t.site u.ext]:req-line)
-          [%'~dojo' *]      (get-file-at /app/landscape [t.site u.ext]:req-line)
-          [%'~groups' *]    (get-file-at /app/landscape [t.site u.ext]:req-line)
-          [%'~landscape' %js %session ~]
-        (some (as-octt:mimes:html "window.ship = '{+:(scow %p our.bowl)}';"))
-      ::
-          [%'~landscape' *]
-        (get-file-at /app/landscape [t.site u.ext]:req-line)
+    ?+  site.req-line  (get-custom req-line)
+    ::
+        [%'' %'index' ~]
```

Comment:
You probably want to `$?([%'' %index ~] [%'~link' *] etc)` all of these cases, so you don't duplicate their logic.

Curious about what the reasoning is for hardcoding these for the "core" apps, instead of having them set this up (like the custom case) during `+on-init`.

Reply:
Fair point about eliminating duplicate logic.

wrt to hardcoding, maybe @loganallenc can speak to that, that was a part I left unchanged from the old version. Was there any reason for hardcoding other than the fact that we just didn't have the ability to dynamically set these yet?

Reply:
Nope, no reason for hardcoding other than lack of dynamicity. I was gonna tear that out and just have on-init set up all the bindings